includes/reader-activation/sync/class-esp-sync.php

```diff
@@ -110,6 +111,19 @@ public static function sync( $contact, $context = '' ) {
 
 		$contact = Sync\Metadata::normalize_contact_data( $contact );
 
+		// Go over all values in the contact's metadata and if they look like dates, then convert them from UTC to the site's timezone.
+		if ( ! empty( $contact['metadata'] ) ) {
+			$timezone     = \wp_timezone();
+			$utc_timezone = new \DateTimeZone( 'UTC' );
+			foreach ( $contact['metadata'] as $key => $value ) {
+				if ( \is_string( $value ) && \preg_match( '/^\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}$/', $value ) ) {
```

Comment:
Maybe this is from past JS-based trauma, but I always shy away from using regex if I can find an alternative solution with other native functions/methods. For this check, since we know the date format we're setting in the `Metadata` class, would it be safer to use something like this to check whether the string is a date? 

```suggestion
				if ( is_string( $value ) && false !== \DateTime::createFromFormat( Metadata::DATE_FORMAT, $value ) ) {
```

Also, this is purely subjective coding style and we're admittedly not super consistent about it throughout our codebase, but we generally try to use the `\` global namespace only for non-native-PHP functions, such as global WP or Woo stuff, and classes that would otherwise throw a namespacing error (such as `DateTime` in this case). Since `is_string` is a [native function](https://www.php.net/manual/en/function.is-string.php), we'd just just say `is_string`. Does it make sense? I'm not sure, but it's how most of us on @Automattic/newspack-product  do it. 😆